### PR TITLE
New package: trexio

### PR DESCRIPTION
**Pull request by q-posev**
Spack specification for the TREXIO library in C.

### CODE REVIEW

**Comment by q-posev** [comment]
@scemama this PR is ready for review

File: var/spack/repos/builtin/packages/trexio/package.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
+# Copyright 2013-2021 Lawrence Livermore National Security, LLC and other
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/trexio/package.py]:
```suggestion
# Copyright 2013-2022 Lawrence Livermore National Security, LLC and other
```

File: var/spack/repos/builtin/packages/trexio/package.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+# To install trexio package in the current envionment
+#
+#     spack install trexio
+#
+# You can edit this file again by typing:
+#
+#     spack edit trexio
+#
+# See the Spack documentation for more information on packaging.
+# ----------------------------------------------------------------------------
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/trexio/package.py]: This is how-to that should be removed.

```suggestion
```

File: var/spack/repos/builtin/packages/trexio/package.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    git      = "https://github.com/TREX-CoE/trexio.git"
+    url      = "https://github.com/TREX-CoE/trexio/releases/download/v2.2.0/trexio-2.2.0.tar.gz"
+
+    # notify when the package is updated.
+    maintainers = ['q-posev', 'scemama']
+
+    version('master', branch='master')
+    version('2.0.0', sha256='6eeef2da44259718b43991eedae4b20d4f90044e38f3b44a8beea52c38b14cb4')
+    version('2.1.0', sha256='232866c943b98fa8a42d34b55e940f7501634eb5bd426555ba970f5c09775e83')
+    version('2.2.0', sha256='e6340c424fcea18ae0b643a5707e16005c7576ee21a5aac679fbc132d70b36d9')
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/trexio/package.py]: Newer versions are listed first in Spack.

```suggestion
    version('2.2.0', sha256='e6340c424fcea18ae0b643a5707e16005c7576ee21a5aac679fbc132d70b36d9')
    version('2.1.0', sha256='232866c943b98fa8a42d34b55e940f7501634eb5bd426555ba970f5c09775e83')
    version('2.0.0', sha256='6eeef2da44259718b43991eedae4b20d4f90044e38f3b44a8beea52c38b14cb4')
```

**Review by tldahlgren — CHANGES_REQUESTED** [review]
Confirmed sha256.  There are a few changes that need to be made.

**Comment by q-posev** [comment]
Thank you @tldahlgren for the review! I have added your changes.

**Review by tldahlgren — APPROVED** [review]

**Pull request merged**